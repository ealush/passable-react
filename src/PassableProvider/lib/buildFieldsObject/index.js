import {isAnObject} from '../index';

export default function buildFieldsObject(initialStateObject) {
    return Object.keys(initialStateObject || {}).reduce((fields, current) => {
        fields[current] = fields[current] || {};
        if (isAnObject(initialStateObject[current])) {
            fields[current] = Object.assign({}, initialStateObject[current]);
        } else {
            // silly assumption. need to revise it.
            fields[current].value = initialStateObject[current];
        }
        return fields;
    }, {});
}