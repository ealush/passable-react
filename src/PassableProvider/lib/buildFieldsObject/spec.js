import buildFieldsObject from './index';

describe('Test buildFieldsObject function', () => {

    it('should reduce initial state into an object', () => {
        expect(buildFieldsObject({
            field_1: {
                checked: true,
                value: 'on'
            }
        })).toMatchSnapshot();
    });

    it('should assume value when not an object', () => {
        expect(buildFieldsObject({
            field_1: true
        })).toMatchSnapshot();
    });

    it('should default to an empty object when no arguments passed', () => {
        expect(buildFieldsObject()).toEqual({});
    });

    it('should use default for null and undefined', () => {
        expect(buildFieldsObject(null)).toEqual({});
        expect(buildFieldsObject(undefined)).toEqual({});
    });
});