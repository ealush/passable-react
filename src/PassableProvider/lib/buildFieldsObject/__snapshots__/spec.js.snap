// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test buildFieldsObject function should assume value when not an object 1`] = `
Object {
  "field_1": Object {
    "errors": Array [],
    "value": true,
    "warnings": Array [],
  },
}
`;

exports[`Test buildFieldsObject function should reduce initial state into an object 1`] = `
Object {
  "field_1": Object {
    "checked": true,
    "errors": Array [],
    "value": "on",
    "warnings": Array [],
  },
}
`;
